src/sentry/static/sentry/app/views/performance/trends/utils.tsx

```diff
@@ -408,6 +372,16 @@ export function getSelectedQueryKey(trendChangeType: TrendChangeType) {
   return trendSelectedQueryKeys[trendChangeType];
 }
 
+export function movingAverage(data, index, size) {
+  const halfSize = Math.round(size / 2);
+  return (
+    data
+      .slice(index - halfSize, index + halfSize)
+      .map(a => a.value)
+      .reduce((a, b) => a + b, 0) / size
+  );
+}
```

Comment:
This is an odd way of calculating the moving average. 

For a given index `index`, the previous `size` data points should be used for calculating the average at the current index. The next `halfSize` data points shouldn't be used for the average.

Reply:
Agree, data from the 'future' shouldn't be included in a previous point's average. The average value for index=5 shouldn't include data from indexes 6-9.

Reply:
🤔 In hindsight I'm not sure why I did this, :+1: updating it to the expected calculation